### DISCUSSION TITLE
How to trigger event on first successful merge by authors using github action?

### DISCUSSION THREAD

**Discussion by theSwapnilSaste** [community/community#26918] [Actions]
<p>I want to append the name of every user on their first successful PR merge. My current understanding is -</p>
<pre><code class="lang-auto">name: Add to Contributors
on:
  push:
    branches:
    - master

jobs:
  comment:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/first-interaction@v1
      with:
        repo-token: ${{ secrets.GITHUB_TOKEN }}
        pr-message: 'Adding this new user to Contributors...'
</code></pre>
<p>Here I want to take output in some variable like isFirst and username and then create a pull request if isFirst is true;<br>
How should I proceed? is there any alternative way?</p>

**Comment by NicoNBW** [comment, marked as answer]
<p><a class="mention" href="/theswapnilsaste">@theSwapnilSaste</a> Thanks for your feedback.<br>
Through the action code, we find that this variable( <code>firstContribution</code>) is checked in the <a href="https://github.com/actions/first-interaction/blob/6947fba47922a25ea543f10f00d99071f6600e82/lib/main.js#L55" rel="nofollow noopener">main.js line55</a>. Based on your needs, you could try to append username when you add message. <a href="https://github.com/NicoNBW/testfirst3/blob/6b5001d425af34e2998a3311e149f3e8d0e4716a/.github/workflows/blank.yml" rel="nofollow noopener">Such as</a>:</p>
<pre><code class="lang-auto">jobs:
  comment:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/first-interaction@v1
        with:
          repo-token: ${{ secrets.TEST_TOKEN }}
          pr-message: 'Adding this ${{ github.event.pull_request.user.login }} to Contributors...'
          issue-message: 'Adding this ${{ github.event.issue.user.login }} to Contributors...'
</code></pre>
<p>The result:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/7/715e573b2e7571a7621745ce06dd3f45eacec7ab.png" data-download-href="/uploads/short-url/gaU4QTY8YUD8981qHPTu3peThj5.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/46212442/181100621-f43cc4ec-3793-453c-9f55-a34507769452.png" alt="image" data-base62-sha1="gaU4QTY8YUD8981qHPTu3peThj5" width="690" height="436" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/7/715e573b2e7571a7621745ce06dd3f45eacec7ab_2_690x436.png, https://aws1.discourse-cdn.com/github/original/2X/7/715e573b2e7571a7621745ce06dd3f45eacec7ab.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/7/715e573b2e7571a7621745ce06dd3f45eacec7ab.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/7/715e573b2e7571a7621745ce06dd3f45eacec7ab_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">979×619 34.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by theSwapnilSaste** [comment]
<p><a class="mention" href="/niconbw">@niconbw</a>  Can you tell how to add this username to README.md at the end in append mode ? I am new with actions.</p>

**Comment by NicoNBW** [comment]
<p>Hi <a class="mention" href="/theswapnilsaste">@theSwapnilSaste</a><br>
Please try to use git command to update README.md. Such as:</p>
<pre><code class="lang-auto">name: Add to Contributors
on:
  pull_request:
    branches:
    - master

jobs:
  comment:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@master
   
    - name: Modify README.md
      run: |
        echo '&lt;br&gt; Adding this ${{ github.event.pull_request.user.login }} to Contributors' &gt;&gt; README.md
        
    - name: Commit files
      run: |
          git config --local user.email "test@github.com"
          git config --local user.name "GitHub Action test"
          git add ./README.md
          git commit -m "Add Contributors"
    - name: Push changes # push the README.md to your repo
      uses: ad-m/github-push-action@master
      with:
        branch: master #ignore if your branch is master
        github_token: ${{ secrets.GITHUB_TOKEN }}
        force: true
</code></pre>

**Comment by NicoNBW** [comment]
<p>Hi <a class="mention" href="/theswapnilsaste">@theSwapnilSaste</a><br>
Did the git command meet your needs? Feel free to contact me if there is any questions related to this ticket.</p>
